app/models/miq_ae_namespace.rb

```diff
@@ -22,16 +22,20 @@ class MiqAeNamespace < ApplicationRecord
   alias ae_namespaces children
   virtual_has_many :ae_namespaces
 
+  before_save :set_fqname
+  virtual_belongs_to :domain
+
   def self.lookup_by_fqname(fqname, include_classes = true)
     return nil if fqname.blank?
 
-    fqname = fqname[0] == '/' ? fqname : "/#{fqname}"
-    fqname = fqname.downcase
-    last = fqname.split('/').last
-    low_name = arel_table[:name].lower
+    low_domain = arel_table[:domain_name].lower
+    low_namespace = arel_table[:relative_path].lower
```

Comment:
These temporary variables don't look necessary...that is, if they are embedded directly below, it would make the below just as, or even more, readable.